src/Elastic.Apm/Api/Tracer.cs

```diff
@@ -206,5 +208,74 @@ public Task<T> CaptureTransaction<T>(string name, string type, Func<Task<T>> fun
 
 			transaction.End();
 		}, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+
+		public void CaptureError(string message, string culprit, StackFrame[] frames = null, string parentId = null,
+			Dictionary<string, Label> labels = null
+		)
+		{
+			var currentTransaction = CurrentExecutionSegmentsContainer.CurrentTransaction;
+
+			IExecutionSegment currentExecutionSegment = CurrentExecutionSegmentsContainer.CurrentSpan;
+			currentExecutionSegment ??= currentTransaction;
```

Comment:
Although it's more verbose, getting the current transaction should be performed only if there's no current span, so that the work is not performed when there is a current span.

Reply:
If there is a current span, we still need the current transaction as well. In the `ExecutionSegmentCommon.CaptureX` methods one parameter is `IExecutionSegment executionSegment` and another one is `Transaction transaction`. 

We need the `IExecutionSegment` (which is either a span or a transaction) to set the `parentId`, and the `Transaction` to set the `transactionId` and to copy the transaction context at the point when the error is captured. So we definitely need that `currentTransaction` variable.

Reply:
Ah, right, I see it's used below